### PR TITLE
pygitup 1.5.0 (new formula)

### PR DESCRIPTION
**Pull request by michaelhood**

### CODE REVIEW

File: Formula/git-up.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    url "https://files.pythonhosted.org/packages/8a/48/a76be51647d0eb9f10e2a4511bf3ffb8cc1e6b14e9e4fab46173aa79f981/termcolor-1.1.0.tar.gz"
+    sha256 "1d6d69ce66211143803fbc56652b41d73b4a400a2891d7bf7a1cdf4c02de613b"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    system bin/"git-up", "--version"
```
Comment (commitay) [inline Formula/git-up.rb]: The test needs to exercise and verify the functionality of the software.

**Comment by michaelhood** [comment]
Sorry for letting this go stale; I'm not sure how to provide a functional test of this software. Doing so would involve cloning a git repo, mutating filesystem state, etc. If this is the desired scope of tests for homebrew-core formulae.. I can give that a shot.

File: Formula/git-up.rb (hunk 1 above)
Comment (michaelhood) [reply Formula/git-up.rb]: Sorry for the late response. Doing so would involve cloning a git repo, mutating filesystem state, etc. Is this desired in a test? If so, can you point me to another formula that does this sort of test (involving filesystem mutation, at least) so I can follow best practices for temp files/cleanup/etc?

Comment (commitay) [reply Formula/git-up.rb]: The test block runs in a sandbox so cleanup isn't an issue.

Comment (DomT4) [reply Formula/git-up.rb]:
```diff
diff --git a/Formula/git-up.rb b/Formula/git-up.rb
index 194aa0fd54..1fe70efc23 100644
--- a/Formula/git-up.rb
+++ b/Formula/git-up.rb
@@ -48,6 +48,9 @@ class GitUp < Formula
   end
 
   test do
-    system bin/"git-up", "--version"
+    system "git", "clone", "https://github.com/Homebrew/install.git"
+    cd "install" do
+      assert_match "Fetching origin", shell_output("#{bin}/git-up")
+    end
   end
 end
```
Should work fine. That repo is unlikely to go anywhere anytime soon.

**Comment by fxcoudert** [comment]
We already have a `gitup` formula, so `git-up` seems too close :(
@michaelhood how would you feel about `pygitup`?

**Comment by michaelhood** [comment]
Sure; I'm not the author, but I don't have any objection to that. Thanks for taking this @fxcoudert!

**Comment by DomT4** [comment]
> how would you feel about pygitup?

This feels reasonable to me FWIW.

**Comment by ilovezfs** [comment]
An alternative would be renaming `gitup` to `git-repo-updater` and then letting this take the `git-up` name, which is what it's called in [PyPI](https://pypi.org/project/git-up/) but `pygitup` seems fine given the upstream docs refer to it as `PyGitUp`.

**Review by fxcoudert — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks again @michaelhood

**Comment by michaelhood** [comment]
Thanks all!

**Pull request closed without merging**